### PR TITLE
Workaround extremely long backoff requested by Azure server

### PR DESCRIPTION
**Pull request by cberner**

### CODE REVIEW

File: autoscaler/azure.py [hunk 1]
```diff
@@ -58,6 +60,51 @@ def add_done_callback(self, fn):
                 self._callbacks.append(fn)
 
 
+# XXX: Azure sometimes sends us a Retry-After: 1200, even when we still have quota, causing our client to appear to hang.
```
Comment (8enmann) [inline autoscaler/azure.py]: why xxx?

File: autoscaler/azure.py [hunk 2]
```diff
@@ -58,6 +60,51 @@ def add_done_callback(self, fn):
...
+# Ignore them and just retry after 30secs
+_RETRY_TIME_LIMIT = 30
+
+
+class AzureRetry(ClientRetry):
```
Comment (8enmann) [inline autoscaler/azure.py]: could call this something more general like OverrideRetry. Move comments above into docstring for this class?

File: autoscaler/azure.py [hunk 3]
```diff
@@ -58,6 +60,51 @@ def add_done_callback(self, fn):
...
+                       'x-ms-ratelimit-remaining-subscription-writes',
+                       'x-ms-ratelimit-remaining-tenant-reads',
+                       'x-ms-ratelimit-remaining-tenant-writes',
+                       'x-ms-ratelimit-remaining-subscription-resource-requests',
+                       'x-ms-ratelimit-remaining-subscription-resource-entities-read',
+                       'x-ms-ratelimit-remaining-tenant-resource-requests',
+                       'x-ms-ratelimit-remaining-tenant-resource-entities-read']:
+            value = response.getheader(header)
+            if value is not None:
+                headers[header] = value
```
Comment (8enmann) [inline autoscaler/azure.py]: break?

File: autoscaler/azure.py (hunk 1 above)
Comment (cberner) [inline autoscaler/azure.py]: https://en.wikipedia.org/wiki/Comment_(computer_programming)#Tags

File: autoscaler/azure.py (hunk 2 above)
Comment (cberner) [inline autoscaler/azure.py]: It's quite Azure specific (see the logging code), but I'll make it something a bit more descriptive

File: autoscaler/azure.py (hunk 3 above)
Comment (cberner) [inline autoscaler/azure.py]: No, this is filtering out any that are `None`, and they're not ordered

**Pull request merged**